### DISCUSSION TITLE
Consequences of changing WEB_DOMAIN

### DISCUSSION THREAD

**Discussion by OLibutzki** [mastodon/mastodon#19897] [Q&A] [answered]
Hi,

I would like to switch the WEB_DOMAIN of my account. The docs state that it's not recommended to do so, because of remote servers' caches.

Is there a chance to change it anyway? Would it be possible to redirect from the old WEB_DOMAIN to the new one?

**Comment by cham2019** [comment]
I would like to know about this too.  I was testing a new instance, with last version 4.0 , however, since I have done some customizations and tests with a "test.domain.net" as the main mastodon domain name, I just wanted to know if it is possible to change it now to "domain.net" since it is ready to be public and on production status.

**Comment by madeofstown** [comment, marked as answer]
Make the switch to the new domain but make sure the old one still points to the server. There is also another Environmental Variable to set that is something like ALTERNATE_DOMAIN. Set that as the old web domain. After a couple weeks you can probably delete the old domain.

**Reply by madeofstown** [reply]
> Make the switch to the new domain but make sure the old one still points to the server. There is also another Environmental Variable to set that is something like ALTERNATE_DOMAIN. Set that as the old web domain. After a couple weeks you can probably delete the old domain.

(Check https://fediverse.observer and look for your old server's uptime, once that drops below 50% it's probably safe to get rid of the old domain)

**Reply by cham2019** [reply]
Thank you!

**Reply by OLibutzki** [reply]
> Make the switch to the new domain but make sure the old one still points to the server. 

By redirecting the webfinger?

> There is also another Environmental Variable to set that is something like ALTERNATE_DOMAIN. Set that as the old web domain. After a couple weeks you can probably delete the old domain.

Can you explain what the ALTERNATE_DOMAIN is all about? What are the benefits of setting it?

**Reply by cham2019** [reply]
As far as I understood, after reading about "ALTERNATE_DOMAIN" here:  https://docs.joinmastodon.org/admin/config/
seems like an instruction to accept and route those coming requests made to the former domain (the one that was being used by the instance before the change to the new domain).   But this only will work as long as you keep pointing the old domain to the instance. 
In my case, I safely edited the .env file since my instance is new, I have no users yet, but I didn't want to re-install mastodon after my tests and editions.  I just wanted to put the server on production just changing the subdomain "test.domain.net" to the top domain "domain.net".  So I wanted to be sure that nothing internally was going to be broken because of that change.  
But for existent instances, specially those with large number of users, changing a domain is like changing your email address domain.  Your server can still work, but without forwarding the old former domain to the instance your external followers from remote instances will have hard time to locate/synchronize with yours.  I'm not expert on Mastodon, I am still learning how to use it.

**Comment by OLibutzki** [comment]
In general, I appreciate if the consequences of changing the domain is explained more deeply in the docs... 

It's recommend not to do it, but I would like to understand, if it only causes some hickups or causes trouble on the long run.

**Reply by madeofstown** [reply]
As long as the old domain still points to the server there should not be any issues. Issues will arise when someone looks for your username at the old domain, and the old domain no longer exists or is pointing somewhere besides your server. Mastodon is kind of like email over HTTPS. If there isn't an active server at the end of the address then the message isn't getting through.

**Comment by OLibutzki** [comment]
I stumbled upon this old documentation. I don't know why it did not make it into the current docs. Maybe it's outdated?

https://github.com/felx/mastodon-documentation/blob/master/Running-Mastodon/Serving_a_different_domain.md

**Comment by OLibutzki** [comment]
The Webfinger docs are helpful, too.

https://docs.joinmastodon.org/spec/webfinger/
